src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -170,6 +170,10 @@ public boolean isCreated() {
 
         @Override
         public void put(String modelId, KNNEngine knnEngine, byte[] modelBlob, ActionListener<IndexResponse> listener) {
+            if (!isCreated()) {
+                throw new IllegalStateException("Cannot put model in index before index has been initialized");
```

Comment:
Question:  Is it possible Model index not created and we end up with this ```put```? If its the case, why not we create Model index on the first ```put``` and proceed?

Reply:
It is possible. I think that makes sense. Will update.